lib/nostrum/struct/flags.ex

```diff
@@ -0,0 +1,194 @@
+  @typedoc """
+  Team User
+  """
+  @type team_user :: bool
+
+  @typedoc """
+  System user
+  """
+  @type system :: bool
+
+  @typedoc """
+  Bug Hunter (Level 2)
+  """
+  @type bug_hunter_level_2 :: bool
+
+  @typedoc """
+  Verified bot
+  """
+  @type verified_bot :: bool
+
+  @typedoc """
+  Verified developer
+  """
+  @type verified_developer :: bool
+
+  @type flags :: %__MODULE__{
+          staff: staff,
+          partner: partner,
+          hypesquad_events: hypesquad_events,
+          bug_hunter_level_1: bug_hunter_level_1,
+          hypesquad_bravery: hypesquad_bravery,
+          hypesquad_brilliance: hypesquad_brilliance,
+          hypesquad_balance: hypesquad_balance,
+          early_supporter: early_supporter,
+          team_user: team_user,
+          system: system,
+          bug_hunter_level_2: bug_hunter_level_2,
+          verified_bot: verified_bot,
+          verified_developer: verified_developer
+        }
+
+  @type t :: flags
+
+  @flag_values [
+    staff: 1 <<< 0,
+    partner: 1 <<< 1,
+    hypesquad_events: 1 <<< 2,
+    bug_hunter_level_1: 1 <<< 3,
+    hypesquad_bravery: 1 <<< 6,
+    hypesquad_brilliance: 1 <<< 7,
+    hypesquad_balance: 1 <<< 8,
+    early_supporter: 1 <<< 9,
+    team_user: 1 <<< 10,
+    system: 1 <<< 12,
+    bug_hunter_level_2: 1 <<< 14,
+    verified_bot: 1 <<< 16,
+    verified_developer: 1 <<< 17
+  ]
+
+  @doc """
+  Constructs a flag struct based on an integer from the Discord API (either public_flags or flags).
+
+  ## Examples
+
+  ```elixir
+  iex> Nostrum.Struct.Flags.from_integer(131842)
+  %Nostrum.Struct.Flags{
+    bug_hunter_level_1: false,
+    bug_hunter_level_2: false,
+    early_supporter: true,
+    hypesquad_balance: true,
+    hypesquad_bravery: false,
+    hypesquad_brilliance: false,
+    hypesquad_events: false,
+    partner: true,
+    staff: false,
+    system: false,
+    team_user: false,
+    verified_bot: false,
+    verified_developer: true
+  }
+  ```
+  """
+  @spec from_integer(integer()) :: t
+  def from_integer(flag_value) do
+    boolean_list =
+      Enum.map(@flag_values, fn {flag, value} ->
+        {flag, (flag_value &&& value) == value}
+      end)
+
+    struct(__MODULE__, boolean_list)
+  end
+
+  @doc """
+  Convert a flag struct to an integer value.
+
+  ## Examples
+
+  ```elixir
+  iex> my_flags = %Nostrum.Struct.Flags{
+    bug_hunter_level_1: false,
+    bug_hunter_level_2: false,
+    early_supporter: true,
+    hypesquad_balance: true,
+    hypesquad_bravery: false,
+    hypesquad_brilliance: false,
+    hypesquad_events: false,
+    partner: true,
+    staff: false,
+    system: false,
+    team_user: false,
+    verified_bot: false,
+    verified_developer: true
+  }
+  iex> Nostrum.Struct.Flags.to_integer(my_flags)
+  131842
+  ```
+  """
+  @spec to_integer(t) :: integer()
+  def to_integer(flag_struct) do
+    booleans =
+      flag_struct
+      |> Map.to_list()
+      |> List.delete({:__struct__, Nostrum.Struct.Flags})
```

Comment:
I'd recommend using `Map.from_struct/1`.

Reply:
Addressed in 2d4d439c6b77c8f2c6179a9b43a94184e93b9a28